dbfit-java/core/src/main/java/dbfit/api/DbQuery.java

```diff
@@ -0,0 +1,32 @@
+package dbfit.api;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+/**
+ * An object representing an executable database operation.
+ * Executing the operation returns a ResultSet result.
+ *
+ */
+public interface DbQuery extends AutoCloseable {
```

Comment:
Ok. So it looks like a good idea to keep `DbCommand` and `DbQuery` separate (segregated) and we couldn't even use the `execute()` method for statements that return a `ResultSet` and those that don't because of the different method signatures.